src/models/util/ParseCalChartUtils.ts

```diff
@@ -0,0 +1,93 @@
+export function ReadFourCharCode(buffer: DataView, offset: number): string {
+  return String.fromCharCode(
+    buffer.getUint8(offset + 0),
+    buffer.getUint8(offset + 1),
+    buffer.getUint8(offset + 2),
+    buffer.getUint8(offset + 3)
+  );
+}
+
+export function ReadTwoCharCode(buffer: DataView, offset: number): string {
+  return String.fromCharCode(
+    buffer.getUint8(offset + 0),
+    buffer.getUint8(offset + 1)
+  );
+}
+
+export function ReadInt32(buffer: DataView, offset: number): number {
+  return buffer.getInt32(offset);
+}
+
+export function ReadInt16(buffer: DataView, offset: number): number {
+  return buffer.getInt16(offset);
+}
+
+export function ReadStringTillEnd(buffer: DataView, offset: number): string {
+  let count = 0;
+  let found = '';
+  while (offset + count < buffer.byteLength) {
+    const value = buffer.getUint8(offset + count++);
+    if (!value) {
+      break;
+    }
+    found += String.fromCharCode(value);
+  }
+  if (offset + count !== buffer.byteLength) {
+    throw 'String not parsed correctly';
+  }
+  return found;
+}
+
+export function
+ReadArrayOfStringsTillEnd(buffer: DataView, offset: number): string[] {
+  const labels: string[] = [];
+
+  let currentLabel = '';
+  while (offset < buffer.byteLength) {
+    const value = buffer.getUint8(offset++);
+    if (value) {
+      currentLabel += String.fromCharCode(value);
+    } else {
+      labels.push(currentLabel);
+      currentLabel = '';
+    }
+  }
+  if (offset !== buffer.byteLength) {
+    throw 'Error parsing strings from block';
+  }
+  return labels;
+}
+
+// takes a data view and returns an array of tuples of [string, data]
```

Comment:
Since we are using typescript, this information is already in the function definition. I like what was included at the top of `ParseCalChart3` where it describes how the data is formatted.
```suggestion
/** 
 * Each chunk is formatted as such:
 * <4char> <size_of_data> <data...> <"END "> <4char>
 * The beginning and end 4char describes what the data represents.
 */
```

Reply:
Done!